3 feat add route to create a product

Cletox opened the pull request:
checkout for updates:

* CRUD products added.
* data model updated.

Cletox committed:
attaching folder structure, productController and productFunction and i updated product data model from db/productsModel

Cletox committed:
rest of CRUD methods added

Cletox committed:
added 'storage' crud, updated pagination functionality to get data bt filter or get all documetns if filter passed as empty. Get method now works by filter, funcionality updated

odarotto commented on src/controllers/productController.js:
```diff
@@ -0,0 +1,108 @@
+  try {
+    const { name, model, brand, categoryId } = req.body;
+    const productData = { name, model, brand, categoryId };
+    //const productData = req.body; // Alternative2
```
These 2 lines could be one

Cletox replied:
every controller was mofied.

odarotto commented on src/controllers/productController.js:
```diff
@@ -0,0 +1,108 @@
+
+
+exports.getAllProducts = async (req, res) => {
+    try {
```
Fix indenting on this function

Cletox replied:
format was applied.

odarotto commented:
Needs unit testing

Cletox committed:
added some controller,functions and paginations. some messagges from errors have been swapped. several instances were corrected. and few endpoint were created.

Cletox committed:
updates

odarotto committed:
fix(all-the-project): fixing the things that Parra added

Parra added a few things without reading the proper documentation, this leaded to the failure of the whole system. Proper fixes were applied.

Cletox committed:
adding updates

Cletox committed:
functions and pagination folders eliminated by innecesaary use case. products service updated with validation, agreggation pipeline process to retrieve filtered and optimizated matching data.

Cletox committed:
forcing updates
